### PR TITLE
Invalidate restores based on non-existent backups

### LINKED ISSUE #246: Bug: $ark restore create <non-existent-backup-name> creates successfully
Should ideally throw an error?

### PR DESCRIPTION
**Pull request by nrb**
Fixes #246

### CODE REVIEW

File: pkg/controller/restore_controller.go [hunk 1]
```diff
@@ -294,6 +294,9 @@ func (controller *restoreController) getValidationErrors(itm *api.Restore) []str
 
 	if itm.Spec.BackupName == "" {
 		validationErrors = append(validationErrors, "BackupName must be non-empty and correspond to the name of a backup in object storage.")
+	} else if _, err := controller.fetchBackup(controller.bucket, itm.Spec.BackupName); err != nil {
+
```
Comment (ncdc) [inline pkg/controller/restore_controller.go]: Remove blank line

**Comment by ncdc** [comment]
Just the 1 comment, then squash please

File: pkg/controller/restore_controller_test.go [hunk 2]
```diff
@@ -182,8 +182,9 @@ func TestProcessRestore(t *testing.T) {
 			name:                        "restore with non-existent backup name fails",
 			restore:                     arktest.NewTestRestore("foo", "bar", api.RestorePhaseNew).WithBackup("backup-1").WithIncludedNamespace("ns-1").Restore,
 			expectedErr:                 false,
-			expectedPhase:               string(api.RestorePhaseInProgress),
 			expectedRestoreErrors:       1,
```
Comment (ncdc) [inline pkg/controller/restore_controller_test.go]: You can remove this - it's actually 0, and the test logic short-circuits before evaluating this value

Comment (nrb) [reply pkg/controller/restore_controller_test.go]: Done

**Pull request merged**